static/app/views/alerts/rules/metric/presets.tsx

```diff
@@ -1,269 +1,204 @@
-};
+async function getHighestVolumeTransaction(
+  client: Client,
+  organizationSlug: string,
+  projectId: string
+): Promise<[string, number] | null> {
+  const result = await client.requestPromise(
+    `/organizations/${organizationSlug}/events/`,
+    {
+      method: 'GET',
+      data: {
+        statsPeriod: '30d',
+        project: projectId,
+        field: ['count()', 'transaction'],
+        sort: '-count',
+        referrer: 'alert.presets.highest-volume',
+        query: 'event.type:transaction',
+        per_page: 1,
+      },
+    }
+  );
+  const transaction = result.data[0];
+  if (transaction) {
+    return [transaction.transaction, transaction['count()']];
+  }
+  return null;
+}
 
 export const PRESET_AGGREGATES: Preset[] = [
   {
-    name: t('Error count'),
-    match: /^count\(\)/,
-    validDataset: [Dataset.ERRORS],
-    default: 'count()',
-    /**
-     * Simple "Open in Discover" button
-     */
-    makeCtaParams: makeDefaultCta,
+    id: 'p95-highest-volume',
+    title: t('Super slow transactions'),
+    description: 'Know when important transactions deviate significantly from normal',
+    Icon: IconGlobe,
+    alertType: 'trans_duration',
+    async makeContext(client, project, organization) {
+      const transaction = (
+        await getHighestVolumeTransaction(client, organization.slug, project.id)
+      )?.[0];
+      return {
+        name: t('p95 Alert for %s', [project.slug]),
+        aggregate: 'p95(transaction.duration)',
+        dataset: Dataset.TRANSACTIONS,
+        eventTypes: [EventTypes.TRANSACTION],
+        timeWindow: 60,
+        comparisonDelta: 1440,
+        comparisonType: AlertRuleComparisonType.CHANGE,
+        thresholdType: AlertRuleThresholdType.ABOVE,
+        triggers: [
+          {
+            label: AlertRuleTriggerType.CRITICAL,
+            alertThreshold: 200,
+            actions: project.teams.slice(0, 4).map(team => ({
+              type: ActionType.EMAIL,
+              targetType: TargetType.TEAM,
+              targetIdentifier: team.id,
+              unsavedDateCreated: new Date().toISOString(),
+              inputChannelId: null,
+              options: null,
+              unsavedId: uniqueId(),
+            })),
+          },
+          {
+            label: AlertRuleTriggerType.WARNING,
+            alertThreshold: 100,
+            actions: [],
+          },
+        ],
+        query: 'transaction:' + transaction,
+      };
+    },
   },
   {
-    name: t('Users affected'),
-    match: /^count_unique\(user\)/,
-    validDataset: [Dataset.ERRORS],
-    default: 'count_unique(user)',
-    /**
-     * Simple "Open in Discover" button
-     */
-    makeCtaParams: makeDefaultCta,
+    id: 'throughput-highest-volume',
+    title: t('Throttled throughput'),
+    description: 'Know when throughput for important transactions drops near 0',
+    Icon: IconGraph,
+    alertType: 'throughput',
+    async makeContext(client, project, organization) {
```

Comment:
@Neo-Zhixing can we shore more code with the other `makeContext` function?

Reply:
Did some abstraction